components/esp_lvgl_port/src/lvgl9/ppa/bsp_dma2d.c

```diff
@@ -0,0 +1,170 @@
+    dma2d_descriptor_t *rx_link_buffer = (dma2d_descriptor_t *)heap_caps_aligned_calloc(64, 1, 64, MALLOC_CAP_SPIRAM);
+    assert(tx_link_buffer);
+    assert(rx_link_buffer);
+    tx_dsc = (dma2d_descriptor_t *)((uint32_t)tx_link_buffer);
+    rx_dsc = (dma2d_descriptor_t *)((uint32_t)rx_link_buffer);
+
+    dma2d_csc_config_t m2m_dma2d_tx_csc = {0};
+    uint16_t src_pbyte = src->format.pixel_format == COLOR_PIXEL_RGB888 ? 3 : 2;
+    uint16_t dest_pbyte = dest->format.pixel_format == COLOR_PIXEL_RGB888 ? 3 : 2;
+
+    if (src->format.pixel_format == COLOR_PIXEL_RGB888 && dest->format.pixel_format == COLOR_PIXEL_RGB565) {
+        m2m_dma2d_tx_csc.tx_csc_option = DMA2D_CSC_TX_RGB888_TO_RGB565;
+    } else if (src->format.pixel_format == COLOR_PIXEL_RGB565 && dest->format.pixel_format == COLOR_PIXEL_RGB888) {
+        m2m_dma2d_tx_csc.tx_csc_option = DMA2D_CSC_TX_RGB565_TO_RGB888;
+    }
+
+    uint8_t *src_alignd = src->pic_buf + src->offset_y * src_pbyte * src->pic_w;
+    uint32_t src_size_alignd = src->pic_w * src->block_h * src_pbyte;
+    make_addr_and_size_align(&src_alignd, &src_size_alignd);
+
+    uint8_t *dest_alignd = dest->pic_buf + dest->offset_y * dest_pbyte * dest->pic_w;
+    uint32_t dest_size_alignd = dest->pic_w * dest->block_h * dest_pbyte;
+    make_addr_and_size_align(&dest_alignd, &dest_size_alignd);
+
+    // Writeback TX buffers and Invalidate RX buffers
+    esp_cache_msync((void *)src_alignd, src_size_alignd, ESP_CACHE_MSYNC_FLAG_DIR_C2M);
+    esp_cache_msync((void *)dest_alignd, dest_size_alignd, ESP_CACHE_MSYNC_FLAG_INVALIDATE);
+
+    SemaphoreHandle_t counting_sem = xSemaphoreCreateCounting(1, 0);
+
+    dma2d_transfer_ability_t transfer_ability_config = {
+        .data_burst_length = DMA2D_DATA_BURST_LENGTH_128,
+        .desc_burst_en = true,
+        .mb_size = DMA2D_MACRO_BLOCK_SIZE_NONE,
+    };
+
+    // DMA description preparation
+    dma2d_link_dscr_init((uint32_t *)tx_dsc, NULL, (void *)src->pic_buf,
+                            src->pic_w, src->pic_h,
+                            src->block_w, src->block_h,
+                            1, 1, dma2d_desc_pixel_format_to_pbyte_value(src->format),
+                            DMA2D_DESCRIPTOR_BLOCK_RW_MODE_SINGLE, src->offset_x, src->offset_y);
+    dma2d_link_dscr_init((uint32_t *)rx_dsc, NULL, (void *)dest->pic_buf,
+                            dest->pic_w, dest->pic_h,
+                            dest->block_w, dest->block_h,
+                            0, 1, dma2d_desc_pixel_format_to_pbyte_value(dest->format),
+                            DMA2D_DESCRIPTOR_BLOCK_RW_MODE_SINGLE, dest->offset_x, dest->offset_y);
+    // Writeback the DMA descriptors
+    esp_cache_msync((void *)tx_dsc, 64, ESP_CACHE_MSYNC_FLAG_DIR_C2M | ESP_CACHE_MSYNC_FLAG_UNALIGNED);
+    esp_cache_msync((void *)rx_dsc, 64, ESP_CACHE_MSYNC_FLAG_DIR_C2M | ESP_CACHE_MSYNC_FLAG_UNALIGNED);
```

Comment:
Same here, you assume cache line size == 64. But it can be 128 too.

Is the unaligned sync sage here? If yes, please provide better comment